### PR TITLE
transport socket: hash SAN transport socket options

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

### CODE REVIEW

File: test/common/upstream/cluster_manager_impl_test.cc [hunk 1]
```diff
@@ -1833,6 +1833,11 @@ TEST_F(ClusterManagerImplTest, DynamicHostRemoveWithTls) {
   ON_CALL(example_com_context, upstreamTransportSocketOptions())
       .WillByDefault(Return(std::make_shared<Network::TransportSocketOptionsImpl>("example.com")));
 
+  NiceMock<MockLoadBalancerContext> example_com_context_with_san;
+  ON_CALL(example_com_context, upstreamTransportSocketOptions())
```
Comment (mattklein123) [inline test/common/upstream/cluster_manager_impl_test.cc]: example_com_context_with_san ? 

Can you make also use a different name for each one (not example.comf for all) just to make sure there isn't accidental overlap?

Comment (lizan) [reply test/common/upstream/cluster_manager_impl_test.cc]: > Can you make also use a different name for each one (not example.comf for all) just to make sure there isn't accidental overlap?

what do you mean? I specifically made this one `example.com` to test it won't colide with the one without SAN.

Comment (mattklein123) [reply test/common/upstream/cluster_manager_impl_test.cc]: I meant that you use "example.com" for both SNI and SAN. Can you add a case where they are different and make sure the we get a different conn pool?

File: test/common/upstream/cluster_manager_impl_test.cc [hunk 2]
```diff
@@ -1833,6 +1833,16 @@ TEST_F(ClusterManagerImplTest, DynamicHostRemoveWithTls) {
   ON_CALL(example_com_context, upstreamTransportSocketOptions())
       .WillByDefault(Return(std::make_shared<Network::TransportSocketOptionsImpl>("example.com")));
 
+  NiceMock<MockLoadBalancerContext> example_com_context_with_san;
+  ON_CALL(example_com_context_with_san, upstreamTransportSocketOptions())
+      .WillByDefault(Return(std::make_shared<Network::TransportSocketOptionsImpl>(
+          "example.com", std::vector<std::string>{"example.com"})));
+
+  NiceMock<MockLoadBalancerContext> example_com_context_with_san2;
+  ON_CALL(example_com_context_with_san, upstreamTransportSocketOptions())
```
Comment (mattklein123) [inline test/common/upstream/cluster_manager_impl_test.cc]: example_com_context_with_san2?

/wait

Comment (lizan) [reply test/common/upstream/cluster_manager_impl_test.cc]: 🤦‍♂  fixed.

**Pull request merged**